tests/e2e/specs/in-game/spectating.spec.js

```diff
@@ -137,6 +140,53 @@ describe('Spectating Games', () => {
     cy.get('.v-overlay').should('not.exist');
   });
 
+  it('Leaves a spectated game and joins another without processing extraneous updates', () => {
+    cy.setupGameAsSpectator();
+    cy.loadGameFixture({
+      p0Hand: [Card.ACE_OF_SPADES],
+      p0Points: [],
+      p0FaceCards: [],
+      p1Hand: [Card.FOUR_OF_CLUBS, Card.ACE_OF_DIAMONDS],
+      p1Points: [Card.ACE_OF_CLUBS],
+      p1FaceCards: [Card.KING_OF_HEARTS],
+    });
+    cy.get('[data-player-hand-card]').should('have.length', 1);
+
+    cy.window().its('cuttle.app.config.globalProperties.$store.state.game').then((game) => {
+      const aceOfSpadesId = getCardId(game, Card.ACE_OF_SPADES);
+      cy.wrap(aceOfSpadesId).as('aceOfSpades');
+    });
+
+    cy.vueRoute('/');
+    cy.signupOpponent(opponentUsername, opponentPassword);
+    cy.setupGameAsP0(true);
+    cy.loadGameFixture({
+      p0Hand: [Card.TWO_OF_CLUBS, Card.TWO_OF_DIAMONDS],
+      p0Points: [],
+      p0FaceCards: [],
+      p1Hand: [Card.TWO_OF_HEARTS, Card.TWO_OF_SPADES],
+      p1Points: [],
+      p1FaceCards: [],
+    });
+    cy.get('[data-player-hand-card]').should('have.length', 2);
+
+    cy.recoverSessionOpponent(playerOne);
+    cy.get('@aceOfSpades').then((aceOfSpadesId) => {
+      cy.playPointsById(aceOfSpadesId);
+    });
+
+    cy.wait(3000);
```

Comment:
I hate this `cy.wait()` but I wasn't sure how to avoid it. The idea here is that after a move happens in the "other" game that we were _previously_ spectating, we want to make sure that our UI _doesn't_ update to reflect that change. The tricky thing there is that cypress assertions run until they pass, so our assertGameState() call will pass erroneously if it runs before the actual problem happens because things were right, but then they weren't.

Waiting long enough for the issue to occur and _then_ asserting the expected game state is my best solution so far for this

Reply:
Could you try catch a state update by element and if it doesn’t happen assert true?

Reply:
The tricky part is that the point of the test is that when an update happens in this no-longer-relevant-to-you game, there is **no** update in your app, either in the store or the UI. So I'm not sure what to check for to ensure that the update went through. Basically it should be when the request that the test command sends for `cy.playPointsById()` gets a response, but there can be a delay between then and the actual socket message, so I can't think of a surefire way to check that the update you're not supposed to see has happened

Reply:
That's an interesting idea. Basically make a knowingly failing assertion. I don't think cypress gives us a way to catch failed assertions though. I'm not sure about the retryability either

Reply:
I'm asking in the cypress discord because it's an interesting technical challenge

Reply:
After an interesting convo, I think the manual `cy.wait()` is the best way to replicate the failure. Keeping it as-is